packages/ui/src/components/JobCard/Details/DetailsContent/JobLogs/JobLogs.tsx

```diff
@@ -1,32 +1,141 @@
-import React, { useEffect, useState } from 'react';
+/* eslint-disable no-console */
+import React, { SyntheticEvent, useEffect, useState, useRef } from 'react';
+import { AppJob } from '@bull-board/api/typings/app';
+import { Button } from '../../../Button/Button';
+import { PlayIcon } from '../../../../Icons/Play';
 import s from './JobLogs.module.css';
 
 interface JobLogsProps {
+  job: AppJob;
   actions: {
     getJobLogs: () => Promise<string[]>;
   };
 }
 
-export const JobLogs = ({ actions }: JobLogsProps) => {
-  const [logs, setLogs] = useState<string[]>([]);
+interface LogsType {
+  lineNumber: number;
+  message: string;
+  isVisible: boolean;
+}
+
+const formatLogs = (logs: string[]): LogsType[] => {
+  return logs.map((message, lineNumber) => ({
+    message,
+    lineNumber,
+    isVisible: true,
+  }));
+};
+
+export const JobLogs = ({ actions, job }: JobLogsProps) => {
+  const pollingTimer = useRef<NodeJS.Timer>();
+  const [logs, setLogs] = useState<LogsType[]>([]);
+  const [liveLogs, setLiveLogs] = useState(false);
+  const [keyword, setKeyword] = useState('');
+  const currentKeyword = useRef(keyword);
+  const newJobId = `${job.name}-${job.id}-logs`;
 
   useEffect(() => {
     let mounted = true;
-    actions.getJobLogs().then((logs) => mounted && setLogs(logs));
+    actions.getJobLogs().then((logs) => {
+      mounted && setLogs(formatLogs(logs));
+    });
+
     return () => {
       mounted = false;
+      if (!!pollingTimer) clearInterval(pollingTimer.current as unknown as number);
     };
   }, []);
 
-  if (!Array.isArray(logs) || !logs.length) {
-    return null;
-  }
+  useEffect(() => {
+    setLogs(getFilteredLogs());
+  }, [keyword]);
+
+  const getLogType = (message: string) => {
+    const msgType = message.match(/((info|warn|error)?):/i)?.[1] || '';
+    return msgType.toLowerCase();
+  };
+
+  const onClickFullScreen = async () => {
+    const el = document.querySelector(`#${newJobId}`) as HTMLElement;
+    if (document.fullscreenElement != el) return await el.requestFullscreen();
+    return document.exitFullscreen();
+  };
+
+  const onClickLiveLogsButton = () => {
+    const el = document.querySelector(`#${newJobId} > div:last-child`) as HTMLElement;
+    const pre = el.querySelector(`pre`) as HTMLElement;
+
+    setLiveLogs(!liveLogs);
+    if (liveLogs && !!pollingTimer) {
+      clearInterval(pollingTimer.current as unknown as number);
+    } else {
+      pollingTimer.current = setInterval(async () => {
+        const logs = await actions.getJobLogs();
+        setLogs(getFilteredLogs(formatLogs(logs)));
+        el.scrollTo({ top: pre.scrollHeight });
+      }, 1000);
```

Comment:
This is too quick, think that it may overload the server.
Server response may be slower that 1 sec.
Maybe usage of `useInterval` (which already exists) will be a better solution.

This value should be configurable from the settings modal.

Reply:
I'll do another `PR` to add these values to settings